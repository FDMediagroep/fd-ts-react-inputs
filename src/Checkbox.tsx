import React, { PureComponent } from "react";
import { createGlobalStyle } from "styled-components";

export interface Props {
    defaultChecked?: boolean;

    /**
     * Mandatory id.
     */
    id: string;

    /**
     * Optional label.
     */
    label?: string;

    /**
     * Optional name used as name for the input field. When omitted the id is used.
     */
    name?: string;

    /**
     * Optional value for the input field. Default is 1.
     */
    value?: string;
}

export default class Checkbox extends PureComponent<Props, any> {
    render() {
        return (
            <>
                <GlobalStyle/>

                <label className="fd-checkbox">
                    <input id={this.props.id} type="checkbox" name={this.props.name ? this.props.name : this.props.id} value={this.props.value} defaultChecked={this.props.defaultChecked}/>
                    <span className="checkbox">{this.props.label}</span>
                </label>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-checkbox {
    input {
        opacity: 0;
        position: absolute;
    }
    .checkbox {
        user-select: none;
        position: relative;
        display: inline-block;
        padding-left: 1.5rem;
        margin-right: 1rem;
        margin-bottom: .5rem;
        cursor: pointer;
        color: #191919;
        font-size: 1rem;
        font-family: 'ProximaNovaRegular', sans-serif;
        &:before {
            content: '';
            position: absolute;
            box-sizing: border-box;
            left: 0;
            top: .25rem;
            width: 1rem;
            height: 1rem;
            border: .0625rem solid #ccbeb3;
            border-radius: .125rem;
            transition: border-color .3s;
            background-color: #fff;
        }
        &:after {
            content: '';
            position: absolute;
            left: .1875rem;
            top: .5rem;
            width: .5rem;
            height: .2rem;
            border-left: .125rem solid #fff;
            border-bottom: .125rem solid #fff;
            opacity: 0;
            transition: all .3s;
            transform: rotate(-45deg);
        }
    }

    input:checked ~ .checkbox {
        &:before {
            background-color: #E1842B;
        }
        &:after {
            opacity: 1;
        }
    }
}
`;

export {GlobalStyle as CheckboxStyle};
