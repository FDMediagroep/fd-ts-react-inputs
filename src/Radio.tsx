import React, { PureComponent } from "react";
import { createGlobalStyle, css } from "styled-components";

export interface Props {
    defaultChecked?: boolean;

    /**
     * Mandatory id.
     */
    id: string;

    /**
     * Optional label.
     */
    label?: string;

    /**
     * Optional name used as name for the input field. When omitted the id is used.
     * When multiple same-named radio inputs exists they will act as a switch.
     */
    name?: string;

    /**
     * Optional value for the input field. Default is 1.
     */
    value?: string;
}

export default function Radio(props: Props) {
    return (
        <>
            <GlobalStyle />
            <label className="fd-input-radio">
                <input
                    id={props.id}
                    type="radio"
                    name={props.name ? props.name : props.id}
                    value={props.value}
                    defaultChecked={props.defaultChecked}
                />
                <span className="radio">{props.label}</span>
            </label>
        </>
    );
}

const styles = css`
    .fd-input-radio {
        line-height: 1.5;
        color: #191919;
        input {
            opacity: 0;
            position: absolute;
        }
        .radio {
            user-select: none;
            position: relative;
            z-index: 1;
            display: inline-block;
            padding-left: 1.5rem;
            margin-right: 1rem;
            margin-bottom: 0.5rem;
            cursor: pointer;
            color: #191919;
            font-size: 1rem;
            line-height: 1.5;
            font-family: "ProximaNovaRegular", sans-serif;
            &:before {
                content: "";
                position: absolute;
                box-sizing: border-box;
                left: 0;
                top: 0.25rem;
                width: 1rem;
                height: 1rem;
                border: 0.0625rem solid #ccbeb3;
                transition: border-color 0.3s;
                background-color: #fff;
                border-radius: 50%;
            }
            &:after {
                content: "";
                position: absolute;
                left: 0.1875rem;
                top: 0.5rem;
                width: 0.5rem;
                height: 0.2rem;
                border-left: 0.125rem solid #fff;
                border-bottom: 0.125rem solid #fff;
                opacity: 0;
                transition: all 0.3s;
                transform: rotate(-45deg);
                border-radius: 50%;
            }
        }

        input:focus ~ .radio {
            &:before {
                border-color: #49a3a1;
            }
        }

        input:checked ~ .radio {
            &:after {
                opacity: 1;
                background-color: #e1842b;
                border: none;
                width: 0.75rem;
                height: 0.75rem;
                left: 0.125rem;
                top: 0.375rem;
            }
        }
    }
`;

const GlobalStyle = createGlobalStyle`${styles}`;

export { styles as RadioStyle };
