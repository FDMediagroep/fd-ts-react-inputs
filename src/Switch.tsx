import React from "react";
import { createGlobalStyle, css } from "styled-components";

export interface Props {
    /**
     * Optional class name(s). Space delimited.
     */
    className?: string;

    /**
     * Optional defaultChecked property is to set the initial state of the switch. Default is false.
     */
    defaultChecked?: boolean;

    /**
     * Mandatory id.
     */
    id: string;

    /**
     * Optional label.
     */
    label?: string;

    /**
     * Optional name is used as name of the input field. When omitted the id is used instead.
     */
    name?: string;

    /**
     * Optional value of the input field. Default is 1.
     */
    value?: string;
}

export default function Switch(props: Props) {
    return (
        <>
            <GlobalStyle/>
            <div className={`fd-input-switch${props.className ? ` ${props.className}` : ''}`}>
                <label className="switch-container">
                    <input type="checkbox" className="switch-input" name={props.name ? props.name : props.id} value={props.value ? props.value : '1'} defaultChecked={props.defaultChecked}/>
                    <span className="switch"/>
                    <span className="switch-label">{props.label}</span>
                </label>
            </div>
        </>
    );
}

const styles = css`
.fd-input-switch {
    input {
        opacity: 0;
        position: absolute;
    }
    .switch {
        position: relative;
        width: 100%;
        height: 100%;
        width: 2.5rem;
        height: 1.5rem;
        display: inline-block;
        margin-right: .5rem;
        margin-bottom: .5rem;
        cursor: pointer;
        &:before {
            content: '';
            position: absolute;
            box-sizing: border-box;
            transition: border-color .3s;
            background-color: #fff;
            transition: all .3s ease-out;
            width: 100%;
            height: 1rem;
            top: .25rem;
            border-radius: .5rem;
            background-color: #f3f3f3;
        }
        &:after {
            content: '';
            position: absolute;
            transition: all .3s ease-out;
            transform: rotate(-45deg);
            width: 1.5rem;
            height: 1.5rem;
            border-radius: 50%;
            box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.2);
            background-color: #999;
            left: 0;
            top: 0;
        }
    }

    input:checked ~ .switch {
        &:before {
            background-color: rgba(225, 132, 43, 0.5);
            border-radius: .5rem;
        }
        &:after {
            left: calc(100% - 1.5rem);
            background-color: #E1842B;
        }
    }

    .switch-label {
        user-select: none;
        cursor: pointer;
        font-size: 1rem;
        color: #191919;
        position: relative;
        top: -1rem;
        font-family: 'ProximaNovaRegular', sans-serif;
    }
}
`;

const GlobalStyle = createGlobalStyle`${styles}`;

export {styles as SwitchStyle};
