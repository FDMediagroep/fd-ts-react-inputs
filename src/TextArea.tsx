import React from 'react';
import { createGlobalStyle, css } from 'styled-components';

declare var localStorage: any;

export interface Props {
    /**
     * Mandatory id of the input element.
     */
    id: string;

    /**
     * Optional name of the input element. Id is used when name is unset.
     */
    name?: string;

    /**
     * Optional className is set as the className of the outmost container element.
     */
    className?: string;

    /**
     * Optional description is set as description text for the input field.
     * Use this to describe the purpose of the field to the user.
     */
    description?: string;

    /**
     * Optional value of the input element.
     */
    value?: string;

    /**
     * Optional. When true the default input validation kicks in. Tip: use in combination with pattern.
     */
    required?: boolean;

    /**
     * Optional. The label shown in the input field.
     */
    label?: string;

    /**
     * Optional. The native browser tooltip text shown on hover. When omitted then the label is used.
     */
    title?: string;

    /**
     * Optional. Custom error message shown when the value does fit the pattern.
     */
    errorMessage?: string;

    /**
     * Optional minimum length of the input field.
     */
    minLength?: number;

    /**
     * Optional maximum length of the input field.
     */
    maxLength?: number;

    /**
     * Optional callback hook triggered when a change event occured in the input field.
     */
    onChange?: (value: React.ChangeEvent<HTMLTextAreaElement>) => void;

    onFocus?: (value: React.FocusEvent<HTMLTextAreaElement>) => void;

    onBlur?: (value: React.FocusEvent<HTMLTextAreaElement>) => void;

    /**
     * Optional placeholder text which is alway shown when the input value is empty. Default value is an empty space ' ' needed for specific styling hacks.
     */
    placeholder?: string;

    /**
     * Optional. Store the value to localStorage which will be used as default value upon subsequent visits of the page.
     * Tip: `id` is used as suffix for the property name.
     */
    localStorage?: boolean;
}

export default class TextArea extends React.PureComponent<Props, any> {
    state: any = {
        errorMessage: this.props.errorMessage || 'Voer een geldige waarde in',
        maxLength: this.props.maxLength || null,
        minLength: this.props.minLength || null,
        required: this.props.required || false,
        value: ''
    };

    componentDidMount() {
        const value = this.props.localStorage ? localStorage.getItem(`fd-textarea-${this.props.id}`) : this.props.value || '';
        if (value) {
            this.setState({
                value
            });
        }
    }

    checkValue = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
        const value = event.target.value;
        if (value) {
            (event.currentTarget as HTMLTextAreaElement).setCustomValidity("");
            this.setState({value});
        } else {
            this.setState({value: ''});
        }
        if (this.props.onChange) {
            this.props.onChange(event);
        }
        if (this.props.localStorage) {
            this.store(value);
        }
    }

    store = (value: string) => {
        if (value) {
            localStorage.setItem(`fd-textarea-${this.props.id}`, value);
        } else {
            localStorage.removeItem(`fd-textarea-${this.props.id}`);
        }
    }

    handleInvalid = (event: React.FormEvent) => {
        (event.currentTarget as HTMLInputElement).setCustomValidity(this.state.errorMessage);
    }

    render() {
        return (
            <>
                <GlobalStyle/>

                <div className={`fd-textarea${this.props.className ? ` ${this.props.className}` : ''}`}>
                    <div className="textarea">
                        <textarea
                            id={this.props.id}
                            name={this.props.name ? this.props.name : this.props.id}
                            minLength={this.state.minLength}
                            maxLength={this.state.maxLength}
                            onChange={this.checkValue}
                            value={this.state.value}
                            required={this.state.required}
                            onInvalid={this.handleInvalid}
                            title={this.props.title ? this.props.title : this.props.label || "Value"}
                            onFocus={this.props.onFocus}
                            onBlur={this.props.onBlur}
                            placeholder={this.props.placeholder ? this.props.placeholder : ' '}
                        />
                        <label htmlFor={this.props.id}>{this.props.label || "Value"} <i className="icon-check"/></label>
                        <div className="textarea-message error-message">{this.props.errorMessage}</div>
                        <div className="textarea-message">{this.props.description}</div>
                    </div>
                </div>
            </>
        );
    }
}

const styles = css`
.fd-textarea {
    display: inline-block;

    i {
        transition: .4s opacity ease-out;
        opacity: 0;
        position: relative;
        color: #3CAB87;
    }

    p {
        font-size: 12px;
        color: #677381;
    }

    .textarea {
        position: relative;
        float: left;
        font-family: 'ProximaNovaRegular',sans-serif;
        width: 100%;
        textarea {
            font-family: 'ProximaNovaRegular',sans-serif;
            padding: 1.4rem 1rem .6rem 1rem;
            font-size: 1rem;
            vertical-align: bottom;
            border: none;
            border-bottom: 0.125rem solid #ccbeb3;
            &:hover {
                border-color: #99908a;
            }
            &:valid {
                border-color: #49a3a1;
            }
            &:focus {
                border: none;
                border-bottom: 0.125rem solid #49a3a1;
            }
            border-radius: 0.1875rem;
            box-sizing: border-box;
            width: 100%;
            height: unset;
            outline: none;
            margin: unset;
            color: #191919;
        }

        label {
            font-family: 'ProximaNovaRegular',sans-serif;
            position: absolute;
            top: 1.2rem;
            left: 1rem;
            line-height: 1;
            padding: 0;
            transition: all .1s ease-out;
            color: #677381;
            font-size: 1rem;
            user-select: none;
            white-space: nowrap;
            cursor: text;
        }

        @supports (-ms-ime-align:auto) {
            textarea ~ label {
                font-size: .75rem;
                top: .5rem;
            }
        }

        textarea:not([placeholder=' ']) ~ label {
            font-size: .75rem;
            top: .5rem;
        }
        textarea:not(:placeholder-shown) ~ label {
            font-size: .75rem;
            top: .5rem;
        }
        textarea:not(:-ms-input-placeholder) ~ label,
        textarea:-ms-input-placeholder:not([placeholder=' ']) ~ label {
            font-size: .75rem;
            top: .5rem;
        }
        textarea:focus ~ label {
            font-size: .75rem;
            top: .5rem;
        }

        .textarea-message {
            transition: all .3s ease-out;
            font-size: .75rem;
            color: #677382;
            line-height: 1;
            font-family: 'ProximaNovaRegular', sans-serif;
            font-weight: normal;
            margin-left: 1rem;
            &:not(.error-message):not(:empty) {
                max-height: 5rem;
                margin: .5rem 1rem;
            }
        }
        .error-message {
            opacity: 0;
            color: #F57268;
            max-height: 0;
            font-family: 'ProximaNovaBold', sans-serif;
        }
    }

    /**
     * textarea validation styles
     */
    textarea:not(:required):focus ~ label > i,
    textarea:required:valid ~ label > i,
    textarea:not(:required):valid:not([placeholder=' ']) ~ label > i {
        opacity: 1;
    }
    textarea:not(:required):not(:placeholder-shown):valid ~ label > i {
        opacity: 1;
    }
    textarea:not(:required):not(:-ms-input-placeholder):valid ~ label > i {
        opacity: 1;
    }

    textarea:required:not(:valid):not(:placeholder-shown):not(:focus) {
        border-color: #F57268;
    }

    textarea:required:not(:valid):not(:placeholder-shown):not(:focus) ~ .error-message:not(:empty) {
        max-height: 5rem;
        opacity: 1;
        margin: .5rem 1rem;
    }
    textarea:required:not(:valid):not(:-ms-input-placeholder):not(:focus) ~ .error-message:not(:empty) {
        max-height: 5rem;
        opacity: 1;
        margin: .5rem 1rem;
    }
}
`;

const GlobalStyle = createGlobalStyle`${styles}`;

export { styles as TextAreaStyle};
