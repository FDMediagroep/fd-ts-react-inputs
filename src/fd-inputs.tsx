import React from 'react';
import { createGlobalStyle } from 'styled-components';

declare var localStorage: any;

export interface Props {
    /**
     * Mandatory id of the input element.
     */
    id: string;

    /**
     * Optional name of the input element. Id is used when name is unset.
     */
    name?: string;

    /**
     * Optional className is set as the className of the outmost container element.
     */
    className?: string;

    /**
     * Optional value of the input element.
     */
    value?: string;

    /**
     * Optional. When true the default input validation kicks in. Tip: use in combination with pattern.
     */
    required?: boolean;

    /**
     * Optional. The pattern to which the input value is validated. Only works when required is set to true.
     * Tip: the exported Patterns enum already contains some validation rules for common use-cases.
     */
    pattern?: Patterns | string;

    /**
     * Optional. The label shown in the input field.
     */
    label?: string;

    /**
     * Optional. The native browser tooltip text shown on hover. When omitted then the label is used.
     */
    title?: string;

    /**
     * Optional. Custom error message shown when the value does fit the pattern.
     */
    errorMessage?: string;

    /**
     * Optional. Type of the input field. Default is `text`.
     */
    type?: string;

    /**
     * Optional minimum length of the input field.
     */
    minLength?: number;

    /**
     * Optional maximum length of the input field.
     */
    maxLength?: number;

    /**
     * Optional callback hook triggered when a change event occured in the input field.
     */
    onChange?: (value: React.ChangeEvent<HTMLInputElement>) => void;

    onFocus?: (value: React.FocusEvent<HTMLInputElement>) => void;

    onBlur?: (value: React.FocusEvent<HTMLInputElement>) => void;

    /**
     * Optional. Store the value to localStorage which will be used as default value upon subsequent visits of the page.
     * Tip: `id` is used as suffix for the property name.
     */
    localStorage?: boolean;
}

export enum Patterns {
    MOBILE = "06[0-9]{8}",
    EMAIL = ".+@.+\\.[a-z]{2,3}"
}

export default class Input extends React.PureComponent<Props, any> {
    state: any = {
        errorMessage: this.props.errorMessage || 'Voer een geldige waarde in',
        hasValue: false,
        maxLength: this.props.maxLength || null,
        minLength: this.props.minLength || null,
        required: this.props.required || false,
        type: this.props.type || 'text',
        value: ''
    };

    componentDidMount() {
        const value = this.props.localStorage ? localStorage.getItem(`fd-input-${this.props.id}`) : this.props.value || '';
        if (value) {
            this.setState({
                hasValue: true,
                value
            });
        }
    }

    checkValue = (event: React.ChangeEvent<HTMLInputElement>) => {
        const value = event.target.value;
        if (value) {
            (event.currentTarget as HTMLInputElement).setCustomValidity("");
            this.setState({hasValue: true, value});
        } else {
            this.setState({hasValue: false, value: ''});
        }
        if (this.props.onChange) {
            this.props.onChange(event);
        }
        if (this.props.localStorage) {
            this.store(value);
        }
    }

    store = (value: string) => {
        if (value) {
            localStorage.setItem(`fd-input-${this.props.id}`, value);
        } else {
            localStorage.removeItem(`fd-input-${this.props.id}`);
        }
    }

    handleInvalid = (event: React.FormEvent) => {
        (event.currentTarget as HTMLInputElement).setCustomValidity(this.state.errorMessage);
    }

    render() {
        return (
            <>
                <GlobalStyle/>

                <div className={`fd-input${this.props.className ? ` ${this.props.className}` : ''}`}>
                    <div className="input">
                        <input
                            className={this.state.hasValue ? 'has-value' : ''}
                            id={this.props.id}
                            name={this.props.name ? this.props.name : this.props.id}
                            type={this.state.type}
                            minLength={this.state.minLength}
                            maxLength={this.state.maxLength}
                            onChange={this.checkValue}
                            value={this.state.value}
                            required={this.state.required}
                            pattern={this.props.pattern}
                            onInvalid={this.handleInvalid}
                            title={this.props.title ? this.props.title : this.props.label || "Value"}
                            onFocus={this.props.onFocus}
                            onBlur={this.props.onBlur}
                            placeholder=" "
                        />
                        <label htmlFor={this.props.id}>{this.props.label || "Value"} <i className="icon-check"/></label>
                    </div>
                </div>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-input {
    display: inline-block;
    i {
        transition: .4s opacity ease-out;
        opacity: 0;
        position: relative;
        color: #3CAB87;
    }
    input:not(:required):not(:placeholder-shown) ~ label > i,
    input:required:valid ~ label > i {
        opacity: 1;
    }
    p {
        font-size: 12px;
        color: #677381;
    }

    .input {
        position: relative;
        float: left;
        font-family: 'ProximaNovaRegular',sans-serif;
        width: 100%;
        input {
            font-family: 'ProximaNovaRegular',sans-serif;
            padding: 1.4rem 1rem .6rem 1rem;
            font-size: 1rem;
            vertical-align: bottom;
            border: none;
            border-bottom: 0.125rem solid #ccbeb3;
            &:hover {
                border-color: #99908a;
            }
            &:valid {
                border-color: #49a3a1;
            }
            &:focus {
                border: none;
                border-bottom: 0.125rem solid #49a3a1;
            }
            border-radius: 0.1875rem;
            box-sizing: border-box;
            width: 100%;
            height: unset;
            outline: none;
            margin: unset;
            color: #191919;
        }
        label {
            font-family: 'ProximaNovaRegular',sans-serif;
            position: absolute;
            top: 1.2rem;
            left: 1rem;
            line-height: 1;
            padding: 0;
            transition: all .1s;
            color: #677381;
            font-size: 1rem;
            user-select: none;
            white-space: nowrap;
            cursor: text;
        }
        input:focus ~ label,
        input.has-value ~ label {
            font-size: .75rem;
            top: .5rem;
        }
    }
}
`;

export { GlobalStyle as StyledInput};
