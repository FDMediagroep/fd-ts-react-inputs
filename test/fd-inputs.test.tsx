import React from 'react';
import {TextInputs} from '../src/fd-inputs';
import {mount, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

describe('FD Input', () => {
    configure({ adapter: new Adapter() });

    it('should render correctly', () => {
        const wrapper = mount(<TextInputs id="input" localStorage={true}/>);
        const instance = (wrapper.instance() as TextInputs);
        instance.store = jest.fn();
        const input = wrapper.find('input');
        input.simulate('change', {target: { value: 'text'}});
        expect(instance.store).toBeCalledWith('text');
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should render with value correctly', () => {
        const wrapper = mount(<TextInputs id="input" value="test value"/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should render with label correctly', () => {
        const wrapper = mount(<TextInputs id="input" label="this is the label" value="test value"/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should render with title correctly', () => {
        const wrapper = mount(<TextInputs id="input" title="this is the title" label="this is the label" value="test value"/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});
