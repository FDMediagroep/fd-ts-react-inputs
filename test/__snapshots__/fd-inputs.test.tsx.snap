// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FD Input should render correctly 1`] = `
<TextInput
  id="input"
  localStorage={true}
>
  <GlobalStyleComponent />
  <div
    className="fd-input"
  >
    <div
      className="input"
    >
      <input
        id="input"
        maxLength={null}
        minLength={null}
        name="input"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Value"
        type="text"
        value="text"
      />
      <label
        htmlFor="input"
      >
        Value
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      />
      <div
        className="input-message"
      />
    </div>
  </div>
</TextInput>
`;

exports[`FD Input should render with label correctly 1`] = `
<TextInput
  id="input"
  label="this is the label"
  value="test value"
>
  <GlobalStyleComponent />
  <div
    className="fd-input"
  >
    <div
      className="input"
    >
      <input
        id="input"
        maxLength={null}
        minLength={null}
        name="input"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="this is the label"
        type="text"
        value="test value"
      />
      <label
        htmlFor="input"
      >
        this is the label
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      />
      <div
        className="input-message"
      />
    </div>
  </div>
</TextInput>
`;

exports[`FD Input should render with title correctly 1`] = `
<TextInput
  id="input"
  label="this is the label"
  title="this is the title"
  value="test value"
>
  <GlobalStyleComponent />
  <div
    className="fd-input"
  >
    <div
      className="input"
    >
      <input
        id="input"
        maxLength={null}
        minLength={null}
        name="input"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="this is the title"
        type="text"
        value="test value"
      />
      <label
        htmlFor="input"
      >
        this is the label
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      />
      <div
        className="input-message"
      />
    </div>
  </div>
</TextInput>
`;

exports[`FD Input should render with value correctly 1`] = `
<TextInput
  id="input"
  value="test value"
>
  <GlobalStyleComponent />
  <div
    className="fd-input"
  >
    <div
      className="input"
    >
      <input
        id="input"
        maxLength={null}
        minLength={null}
        name="input"
        onChange={[Function]}
        onInvalid={[Function]}
        placeholder=" "
        required={false}
        title="Value"
        type="text"
        value="test value"
      />
      <label
        htmlFor="input"
      >
        Value
         
        <i
          className="icon-check"
        />
      </label>
      <div
        className="input-message error-message"
      />
      <div
        className="input-message"
      />
    </div>
  </div>
</TextInput>
`;
